File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-shared-5-4/BusinessUnit.cpp
// BusinessUnit.cpp 
#include "BusinessUnit.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

BusinessUnit::BusinessUnit(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //nameNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* nameNode = xmlNode->FirstChildElement("name");

   if(nameNode){nameIsNull_ = false;}
   else{nameIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- nameNode , address : " << nameNode << std::endl;
   #endif
   if(nameNode)
   {
      if(nameNode->Attribute("href") || nameNode->Attribute("id"))
      {
          if(nameNode->Attribute("id"))
          {
             nameIDRef_ = nameNode->Attribute("id");
             name_ = boost::shared_ptr<XsdTypeString>(new XsdTypeString(nameNode));
             name_->setID(nameIDRef_);
             IDManager::instance().setID(nameIDRef_,name_);
          }
          else if(nameNode->Attribute("href")) { nameIDRef_ = nameNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { name_ = boost::shared_ptr<XsdTypeString>(new XsdTypeString(nameNode));}
   }

   //businessUnitIdNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* businessUnitIdNode = xmlNode->FirstChildElement("businessUnitId");

   if(businessUnitIdNode){businessUnitIdIsNull_ = false;}
   else{businessUnitIdIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- businessUnitIdNode , address : " << businessUnitIdNode << std::endl;
   #endif
   if(businessUnitIdNode)
   {
      if(businessUnitIdNode->Attribute("href") || businessUnitIdNode->Attribute("id"))
      {
          if(businessUnitIdNode->Attribute("id"))
          {
             businessUnitIdIDRef_ = businessUnitIdNode->Attribute("id");
             businessUnitId_ = boost::shared_ptr<Unit>(new Unit(businessUnitIdNode));
             businessUnitId_->setID(businessUnitIdIDRef_);
             IDManager::instance().setID(businessUnitIdIDRef_,businessUnitId_);
          }
          else if(businessUnitIdNode->Attribute("href")) { businessUnitIdIDRef_ = businessUnitIdNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { businessUnitId_ = boost::shared_ptr<Unit>(new Unit(businessUnitIdNode));}
   }

   //contactInfoNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* contactInfoNode = xmlNode->FirstChildElement("contactInfo");

   if(contactInfoNode){contactInfoIsNull_ = false;}
   else{contactInfoIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- contactInfoNode , address : " << contactInfoNode << std::endl;
   #endif
   if(contactInfoNode)
   {
      if(contactInfoNode->Attribute("href") || contactInfoNode->Attribute("id"))
      {
          if(contactInfoNode->Attribute("id"))
          {
             contactInfoIDRef_ = contactInfoNode->Attribute("id");
             contactInfo_ = boost::shared_ptr<ContactInformation>(new ContactInformation(contactInfoNode));
             contactInfo_->setID(contactInfoIDRef_);
             IDManager::instance().setID(contactInfoIDRef_,contactInfo_);
          }
          else if(contactInfoNode->Attribute("href")) { contactInfoIDRef_ = contactInfoNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { contactInfo_ = boost::shared_ptr<ContactInformation>(new ContactInformation(contactInfoNode));}
   }

   //countryNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* countryNode = xmlNode->FirstChildElement("country");

   if(countryNode){countryIsNull_ = false;}
   else{countryIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- countryNode , address : " << countryNode << std::endl;
   #endif
   if(countryNode)
   {
      if(countryNode->Attribute("href") || countryNode->Attribute("id"))
      {
          if(countryNode->Attribute("id"))
          {
             countryIDRef_ = countryNode->Attribute("id");
             country_ = boost::shared_ptr<CountryCode>(new CountryCode(countryNode));
             country_->setID(countryIDRef_);
             IDManager::instance().setID(countryIDRef_,country_);
          }
          else if(countryNode->Attribute("href")) { countryIDRef_ = countryNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { country_ = boost::shared_ptr<CountryCode>(new CountryCode(countryNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<XsdTypeString> BusinessUnit::getName()
{
   if(!this->nameIsNull_){
        if(nameIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeString>(IDManager::instance().getID(nameIDRef_));
        }else{
             return this->name_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeString>();
   }
}
boost::shared_ptr<Unit> BusinessUnit::getBusinessUnitId()
{
   if(!this->businessUnitIdIsNull_){
        if(businessUnitIdIDRef_ != ""){
             return boost::shared_static_cast<Unit>(IDManager::instance().getID(businessUnitIdIDRef_));
        }else{
             return this->businessUnitId_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<Unit>();
   }
}
boost::shared_ptr<ContactInformation> BusinessUnit::getContactInfo()
{
   if(!this->contactInfoIsNull_){
        if(contactInfoIDRef_ != ""){
             return boost::shared_static_cast<ContactInformation>(IDManager::instance().getID(contactInfoIDRef_));
        }else{
             return this->contactInfo_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<ContactInformation>();
   }
}
boost::shared_ptr<CountryCode> BusinessUnit::getCountry()
{
   if(!this->countryIsNull_){
        if(countryIDRef_ != ""){
             return boost::shared_static_cast<CountryCode>(IDManager::instance().getID(countryIDRef_));
        }else{
             return this->country_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<CountryCode>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-shared-5-4/BusinessUnit.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class BusinessUnit : ISerialized
    {
        public BusinessUnit(XmlNode xmlNode)
        {
            XmlNode nameNode = xmlNode.SelectSingleNode("name");
            
            if (nameNode != null)
            {
                if (nameNode.Attributes["href"] != null || nameNode.Attributes["id"] != null) 
                {
                    if (nameNode.Attributes["id"] != null) 
                    {
                        nameIDRef_ = nameNode.Attributes["id"].Value;
                        XsdTypeString ob = new XsdTypeString(nameNode);
                        IDManager.SetID(nameIDRef_, ob);
                    }
                    else if (nameNode.Attributes["href"] != null)
                    {
                        nameIDRef_ = nameNode.Attributes["href"].Value;
                    }
                    else
                    {
                        name_ = new XsdTypeString(nameNode);
                    }
                }
                else
                {
                    name_ = new XsdTypeString(nameNode);
                }
            }
            
        
            XmlNode businessUnitIdNode = xmlNode.SelectSingleNode("businessUnitId");
            
            if (businessUnitIdNode != null)
            {
                if (businessUnitIdNode.Attributes["href"] != null || businessUnitIdNode.Attributes["id"] != null) 
                {
                    if (businessUnitIdNode.Attributes["id"] != null) 
                    {
                        businessUnitIdIDRef_ = businessUnitIdNode.Attributes["id"].Value;
                        Unit ob = new Unit(businessUnitIdNode);
                        IDManager.SetID(businessUnitIdIDRef_, ob);
                    }
                    else if (businessUnitIdNode.Attributes["href"] != null)
                    {
                        businessUnitIdIDRef_ = businessUnitIdNode.Attributes["href"].Value;
                    }
                    else
                    {
                        businessUnitId_ = new Unit(businessUnitIdNode);
                    }
                }
                else
                {
                    businessUnitId_ = new Unit(businessUnitIdNode);
                }
            }
            
        
            XmlNode contactInfoNode = xmlNode.SelectSingleNode("contactInfo");
            
            if (contactInfoNode != null)
            {
                if (contactInfoNode.Attributes["href"] != null || contactInfoNode.Attributes["id"] != null) 
                {
                    if (contactInfoNode.Attributes["id"] != null) 
                    {
                        contactInfoIDRef_ = contactInfoNode.Attributes["id"].Value;
                        ContactInformation ob = new ContactInformation(contactInfoNode);
                        IDManager.SetID(contactInfoIDRef_, ob);
                    }
                    else if (contactInfoNode.Attributes["href"] != null)
                    {
                        contactInfoIDRef_ = contactInfoNode.Attributes["href"].Value;
                    }
                    else
                    {
                        contactInfo_ = new ContactInformation(contactInfoNode);
                    }
                }
                else
                {
                    contactInfo_ = new ContactInformation(contactInfoNode);
                }
            }
            
        
            XmlNode countryNode = xmlNode.SelectSingleNode("country");
            
            if (countryNode != null)
            {
                if (countryNode.Attributes["href"] != null || countryNode.Attributes["id"] != null) 
                {
                    if (countryNode.Attributes["id"] != null) 
                    {
                        countryIDRef_ = countryNode.Attributes["id"].Value;
                        CountryCode ob = new CountryCode(countryNode);
                        IDManager.SetID(countryIDRef_, ob);
                    }
                    else if (countryNode.Attributes["href"] != null)
                    {
                        countryIDRef_ = countryNode.Attributes["href"].Value;
                    }
                    else
                    {
                        country_ = new CountryCode(countryNode);
                    }
                }
                else
                {
                    country_ = new CountryCode(countryNode);
                }
            }
            
        
        }
        
    
        #region Name_
        private XsdTypeString name_;
        public XsdTypeString Name_
        {
            get
            {
                if (this.name_ != null)
                {
                    return this.name_; 
                }
                else if (this.nameIDRef_ != null)
                {
                    name_ = IDManager.getID(nameIDRef_) as XsdTypeString;
                    return this.name_; 
                }
                else
                {
                      return this.name_; 
                }
            }
            set
            {
                if (this.name_ != value)
                {
                    this.name_ = value;
                }
            }
        }
        #endregion
        
        public string nameIDRef_ { get; set; }
        #region BusinessUnitId_
        private Unit businessUnitId_;
        public Unit BusinessUnitId_
        {
            get
            {
                if (this.businessUnitId_ != null)
                {
                    return this.businessUnitId_; 
                }
                else if (this.businessUnitIdIDRef_ != null)
                {
                    businessUnitId_ = IDManager.getID(businessUnitIdIDRef_) as Unit;
                    return this.businessUnitId_; 
                }
                else
                {
                      return this.businessUnitId_; 
                }
            }
            set
            {
                if (this.businessUnitId_ != value)
                {
                    this.businessUnitId_ = value;
                }
            }
        }
        #endregion
        
        public string businessUnitIdIDRef_ { get; set; }
        #region ContactInfo_
        private ContactInformation contactInfo_;
        public ContactInformation ContactInfo_
        {
            get
            {
                if (this.contactInfo_ != null)
                {
                    return this.contactInfo_; 
                }
                else if (this.contactInfoIDRef_ != null)
                {
                    contactInfo_ = IDManager.getID(contactInfoIDRef_) as ContactInformation;
                    return this.contactInfo_; 
                }
                else
                {
                      return this.contactInfo_; 
                }
            }
            set
            {
                if (this.contactInfo_ != value)
                {
                    this.contactInfo_ = value;
                }
            }
        }
        #endregion
        
        public string contactInfoIDRef_ { get; set; }
        #region Country_
        private CountryCode country_;
        public CountryCode Country_
        {
            get
            {
                if (this.country_ != null)
                {
                    return this.country_; 
                }
                else if (this.countryIDRef_ != null)
                {
                    country_ = IDManager.getID(countryIDRef_) as CountryCode;
                    return this.country_; 
                }
                else
                {
                      return this.country_; 
                }
            }
            set
            {
                if (this.country_ != value)
                {
                    this.country_ = value;
                }
            }
        }
        #endregion
        
        public string countryIDRef_ { get; set; }
        
    
        
    
    }
    
}

